### PR TITLE
AF: champion vol-10x seed=42 extended 600-min replication

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The AF champion (PR #3135, vol-weight=10x + EMA=0.999) was still converging at ep763 when the 360-min timeout hit. Jin (#3304) is running extended 600-min with default seed. This run adds seed=42 for verification — different random initialization may find a better optimum, and we need multiple seeds to confirm the champion is robust.

## Instructions

**No code changes needed.** Champion config replication with seed=42 and 600-min budget.

**Trial 1 — seed=42, 600-min:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 \
  --grad-clip 1.0 --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --ema-decay 0.999 \
  --volume-loss-weight 10.0 \
  --save-checkpoint \
  --seed 42 \
  --wandb-group af-extended-champion --wandb-name "spike/af-vol10-seed42-600min" --agent spike
```

**Trial 2 — seed=7, 600-min:**
Same but `--seed 7` and wandb-name `"spike/af-vol10-seed7-600min"`

Report best_val/surface_mse, best_val/vol_mse, and best_test metrics for both. The key question: does the champion improvement compound with more epochs, or does it plateau around ep800?

## Baseline

| Metric | Value | Epoch | W&B |
|--------|-------|-------|-----|
| val_primary/surface_mse | **0.000296** | 704 | sh2zyfwr |
| val_primary/vol_mse | **0.002039** | 743 | sh2zyfwr |

- Config: 2L/256d/4H, AdamW lr=6e-4, T_max=50, gc=1.0, WD=1e-2, EMA=0.999, vol-weight=10x
- Champion still converging at ep763 timeout — extended runs should push lower

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT spike:

## Results

### Trial 1 — seed=42 (W&B: [`yz9wkraj`](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/yz9wkraj))

| Metric | Val (best) | Test | Epoch |
|--------|-----------|------|-------|
| surface_mse | **0.000479** | **0.001387** | 515 |
| surface_mse_p | 0.001906 | 0.005525 | 515 |
| volume_mse | 0.003620 | 0.004554 | 515 |

- Diverged at ~ep600-613: loss → NaN, grad_norm → NaN/Inf, model outputs collapsed to constants (surface_mse frozen at 0.621)
- Pressure component drove the explosion (surface_mse_p went from 0.002 → 2.48)
- Ran all 999 epochs (most post-divergence were garbage)

**Command:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 \
  --grad-clip 1.0 --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --ema-decay 0.999 \
  --volume-loss-weight 10.0 \
  --save-checkpoint \
  --seed 42 \
  --wandb-group af-extended-champion --wandb-name "spike/af-vol10-seed42-600min" --agent spike
```

### Trial 2 — seed=7 (W&B: [`r6rhawkb`](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/r6rhawkb))

| Metric | Val (best) | Test | Epoch |
|--------|-----------|------|-------|
| surface_mse | **0.000433** | **0.000953** | 575 |
| surface_mse_p | 0.001717 | 0.003793 | 575 |
| volume_mse | 0.004097 | 0.004747 | 575 |

- Diverged at ~ep576 (immediately after best epoch): surface_mse jumped from 0.000433 → 0.0006 → 0.001 → 0.007 → 0.093 → 0.175 in ~10 epochs
- Pressure component exploded (surface_mse_p → 0.68+)
- Ran until 600-min timeout (~999 epochs of post-divergence garbage)

**Command:** Same as Trial 1 but `CUDA_VISIBLE_DEVICES=1 --seed 7 --wandb-name "spike/af-vol10-seed7-600min"`

### Comparison vs Baseline

| Metric | Seed=42 | Seed=7 | Baseline (sh2zyfwr) | Δ seed42 | Δ seed7 |
|--------|---------|--------|---------------------|----------|---------|
| val surface_mse | 0.000479 | 0.000433 | **0.000296** | +62% | +46% |
| val volume_mse | 0.003620 | 0.004097 | **0.002039** | +78% | +101% |
| best epoch | 515 | 575 | 704 | — | — |
| diverged at | ~ep600 | ~ep576 | still converging (ep763) | — | — |
| test surface_mse | 0.001387 | 0.000953 | — | — | — |

Peak memory: ~7 GB VRAM per run (H100 97 GB)

### What happened

**The champion improvement does NOT compound — it's seed-fragile.**

Both seeds showed steady improvement through ~500 epochs, with improvements clustering near cosine annealing cycle boundaries (T_max=50). However, both hit catastrophic divergence around ep550-600:

1. **Seed=42**: Gradual vol_mse drift from ep580, then explosive divergence at ep605-613. Loss → NaN, gradients all non-finite, model outputs collapsed to constants.

2. **Seed=7**: Sharper failure — diverged immediately after best epoch (ep575→ep576). Surface_mse increased 400x in ~10 epochs. Pressure field exploded first.

Meanwhile, the baseline (sh2zyfwr, presumably default seed) reached ep763 without diverging and was **still converging**. This means:

- The 10x vol-weight config is **not seed-robust** — 2 out of 3 tested seeds diverge before reaching the baseline's best epoch
- The divergence is **pressure-driven** in both cases — surface_mse_p explodes first, then drags everything else
- The aggressive vol-weight=10x amplifies pressure gradients until numerical instability cascades

Neither seed came close to beating the baseline. Seed=7 got closest (val surface 0.000433 vs 0.000296) but still +46% worse, and diverged 129 epochs before the baseline's best.

### Suggested follow-ups

1. **Lower vol-weight** (e.g., 3x-5x): The 10x amplification appears to be the root cause of instability. A milder weight might allow training to reach ep700+ across seeds without divergence.
2. **Gradient clipping tightening**: Current gc=1.0 isn't preventing the pressure gradient explosion. Trying gc=0.5 or even 0.1 with vol-weight=10x might stabilize.
3. **Learning rate warmup or reduction**: The divergence often happens near cosine cycle peaks (LR = 6e-4). A lower peak LR (3e-4) or warmup period could help.
4. **Early stopping with patience**: Since both seeds diverged suddenly, adding early stopping with patience=50 would save significant compute time on doomed runs.
5. **Wait for Jin's #3304 results** with default seed — if that also diverges, the champion config is fundamentally unstable beyond ep500-600 regardless of seed.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3309 because ADVISOR: Closing — **config error.** Your command used AdamW lr=6e-4 with 2 layers, but the AF champion is Lion lr=1.25e-4 with 4 layers + Lookahead + compile. Results are not comparable to baseline. The divergence analysis is interesting but applies to the wrong config.

AF champion reproduce command for reference:
```
python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model --vol-loss-weight 10
```

Spike reassigned to DM metric-aware sweep..
